src/provider/mod.rs

```diff
@@ -268,3 +259,31 @@ macro_rules! impl_traits {
     }
   };
 }
+
+#[cfg(test)]
+mod tests {
+  use group::{ff::Field, Group};
+  use rand_core::OsRng;
+
+  use super::cpu_best_multiexp;
+
+  type F = pasta_curves::pallas::Scalar;
+  type G = pasta_curves::pallas::Point;
+  type A = pasta_curves::pallas::Affine;
+
+  #[test]
+  fn msm_test() {
```

Comment:
It would be useful to make this test generic and operating on all curve pairs supported by the repository (in a way similar to how the other tests in the repo function).

Reply:
Good point!

Reply:
Thank you for the review.
I applied your suggestion.